src/notation/internal/notationinteraction.cpp

```diff
@@ -2069,18 +2069,16 @@ bool NotationInteraction::selectInstrument(mu::engraving::InstrumentChange* inst
         return false;
     }
 
-    RetVal<InstrumentTemplate> templ = selectInstrumentScenario()->selectInstrument();
-    if (!templ.ret) {
-        return false;
-    }
-
-    Instrument newInstrument = Instrument::fromTemplate(&templ.val);
-    instrumentChange->setInit(true);
-    instrumentChange->setupInstrument(&newInstrument);
+    async::Promise<InstrumentTemplate> templ = selectInstrumentScenario()->selectInstrument();
+    templ.onResolve(this, [this, instrumentChange](const InstrumentTemplate& val) {
+        Instrument newInstrument = Instrument::fromTemplate(&val);
+        instrumentChange->setInit(true);
+        instrumentChange->setupInstrument(&newInstrument);
 
-    if (newInstrument.useDrumset()) {
-        cleanupDrumsetChanges(instrumentChange);
-    }
+        if (newInstrument.useDrumset()) {
+            cleanupDrumsetChanges(instrumentChange);
+        }
+    });
 
     return true;
```

Comment:
In `NotationInteraction::doDropStandard`, after calling `selectInstrument`, we do a lot more (like updating the score, selecting the added element), but now we no longer wait for the instrument to be selected - we just proceed immediately (yes?). I think this is a bug.

same for `applyDropPaletteElement`, `addText`